src/System.Transactions.Local/src/System/Transactions/TransactionInterop.cs

```diff
@@ -118,13 +118,13 @@ public static Transaction GetTransactionFromExportCookie(byte[] cookie)
 
             // Extract the transaction guid from the propagation token to see if we already have a
             // transaction object for the transaction.
-            byte[] guidByteArray = new byte[16];
+            Span<byte> guidByteArray = stackalloc byte[16];
             for (int i = 0; i < guidByteArray.Length; i++)
             {
                 // In a cookie, the transaction guid is preceeded by a signature guid.
                 guidByteArray[i] = cookie[i + 16];
             }
-            Guid txId = new Guid(guidByteArray);
+            var txId = new Guid(guidByteArray);
```

Comment:
Can we bypass _guidByteArray_ entirely and instead write this as follows?

```cs
Guid txId = new Guid(cookie.AsSpan(16, 16))?
```

Reply:
I was thinking of this initially, but I can't seem to be able to reference System.MemoryExtension. This is odd to me, as I am able to reference Span, which have both been added in .net core 2.1, so I am not sure why I can use Span, but not System.MemoryExtensions.
How dependencies in this repository have been rather confusing to me, do you know why?

Reply:
Span is exposed from `System.Runtime` which is being references in this project: https://github.com/dotnet/corefx/blob/7336f57f745e7399803902efca8fd085ff6ca326/src/System.Transactions.Local/src/System.Transactions.Local.csproj#L44-L57

MemoryExtensions is exposed from `System.Memory`, so if you need those APIs here, you would need to add a reference to that:
https://github.com/dotnet/corefx/blob/7336f57f745e7399803902efca8fd085ff6ca326/src/System.Memory/ref/System.Memory.cs#L10

Reply:
Addressed in https://github.com/dotnet/corefx/pull/36838/commits/677b924f461fbcdb6366f45d7844cbc3cbafd6db